src/pybind/mgr/dashboard/controllers/osd.py

```diff
@@ -493,24 +503,26 @@ def deployment_options(self):
         ssds = 0
         nvmes = 0
         res = DeploymentOptions()
-        devices = {}
 
         for inventory_host in orch.inventory.list(hosts=None, refresh=True):
             for device in inventory_host.devices.devices:
                 if device.available:
-                    devices[device.path] = device
                     if device.human_readable_type == 'hdd':
                         hdds += 1
-                    elif device.human_readable_type == 'ssd':
+                    else:
                         ssds += 1
-                    elif device.human_readable_type == 'nvme':
+                        # we still don't know how to infer nvmes
+                        # Tracker: https://tracker.ceph.com/issues/55728
                         nvmes += 1
+
         if hdds:
             res.options[OsdDeploymentOptions.COST_CAPACITY].available = True
             res.recommended_option = OsdDeploymentOptions.COST_CAPACITY
         if ssds:
             res.options[OsdDeploymentOptions.THROUGHPUT].available = True
             res.recommended_option = OsdDeploymentOptions.THROUGHPUT
+        if nvmes:
```

Comment:
@pereman2 all **hdds** , **ssds** and **nvmes** are holding number of respective devices it has, But based on that how is it deciding which osd deployment option to recommend 
Suppose consider cluster has 3 osd nodes in that one host has **ssds** and other two are **nvme** from the code read which osd deployment option it recommend?
Can you please explain the logic on how osd deployment option is recommended?